syntax = "proto3";
package project;




import "pipeline_sdk/model/pipeline/project.proto";
import "pipeline_sdk/model/pipeline/provider.proto";
import "pipeline_sdk/model/pipeline/pipeline.proto";

/*
Get请求
*/
message GetRequest {
    /*
    代码项目id
    */
    string project_id = 1 ;
}

/*
Get返回
*/
message GetResponse {
    /*
    代码项目
    */
    pipeline.Project project = 1 ;
    /*
    project关联的provider信息
    */
    pipeline.Provider provider = 2 ;
    /*
    project关联的pipeline信息
    */
    repeated pipeline.Pipeline pipeline = 3 ;
}

/*
GetApi返回
*/
message GetResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    project.GetResponse data = 4 ;
}

